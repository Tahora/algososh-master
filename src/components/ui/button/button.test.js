import React from 'react';
import renderer from 'react-test-renderer';
import {Button} from './button';
import {render, screen, fireEvent} from '@testing-library/react';


describe('Button test', function () {
        it('with text', () => {
            const tree = renderer
                .create(<Button text="Test text"/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('without text', () => {
            const tree = renderer
                .create(<Button/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('disabled', () => {
            const tree = renderer
                .create(<Button disabled={true}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('with loader', () => {
            const tree = renderer
                .create(<Button isLoader={true}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('onClick callback', () => {
            window.alert = jest.fn();
            const message = 'it\'s work!'
            const callback = () => {
                alert(message)
            }
            render(<Button data-testid="test-btn"
                           onClick={callback}/>)
            const btn = screen.getByTestId("test-btn");
            fireEvent.click(btn);
            expect(window.alert).toHaveBeenCalledWith(message);
        });
    }
)
