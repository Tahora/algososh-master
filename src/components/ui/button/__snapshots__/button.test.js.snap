// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app works correctly with routes Кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`app works correctly with routes Кнопка заблокированная 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`app works correctly with routes Кнопка с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`app works correctly with routes Кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Test text
  </p>
</button>
`;

exports[`Кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка заблокированная 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Test text
  </p>
</button>
`;
